### DISCUSSION TITLE
Header indenting

### DISCUSSION THREAD

**Discussion by AtomsForHire** [nvim-neorg/neorg#602] [Q&A] [answered]
Whenever I'm typing out headers, before hitting enter the indenting is fine. For example
```
* HEAD
    ** HEAD2|
```
but on entering a new line, the header goes back to root
```
* HEAD
** HEAD2
|
```
Not really sure how to fix this, thought treesitter might have been doing something but my treesitter config is pretty simple
```
require('nvim-treesitter.configs').setup {
	ensure_installed = { 'norg' },
	highlight = { enable = true },
}

local os = vim.loop.os_uname().sysname

if ( os == 'Darwin' ) then
	require('nvim-treesitter.install').compilers = { 'gcc-11' }
end

```
and my neorg is also pretty simple
```
local neorg = require("neorg")
neorg.setup{
	load = {
		["core.defaults"] = {},
		["core.norg.completion"] = {
			config = {
				engine = 'nvim-cmp',
			}
		},
		["core.norg.concealer"] = { }
	}
}

```

Edit: The text after headers are still indented properly so it looks like this
```
* TEST
	blah
** TEST
	 blah
*** TEST
		blah

```
( looks better in nvim )

**Comment by katawful** [comment, marked as answer]
Indentation in neorg is literal, not a visual thing like in markdown (where rendering determines indentation)

Each heading level gets 1 space indent over the previous. The proper indentation should be

```
* head
  Text
^^
** head 2
   Text
^^^
```
Note the literal spaces pointed out with `^`

The indentexpr built in covers most cases, and you shouldn't ever need to add white space. Indentation level also isn't meant to be increased without a new header, though there is `---` and `===` for returning to previous indentation and root indentation respectively

**Reply by AtomsForHire** [reply]
Ah alright, so then is the indenting in the 'First class treesitter support' showcase something outside neorg?

**Reply by max397574** [reply]
it's just outdated

**Reply by AtomsForHire** [reply]
Ah alright, that clears things up. Thanks!

**Reply by vhyrro** [reply]
Yeah like max said the gifs are from before the rewrite of the indent engine :)

One thing to note is that indents in norg are *relative*, try doing this:
```
*       Title
```
and press enter, you'll see your text indented the same amount as the title above!
